hikari/internal/routes.py

```diff
@@ -241,53 +248,68 @@ def compile(self, base_url: str, *, file_format: str, size: int | None = None, *
         Raises
         ------
         TypeError
-            If a GIF is requested, but the asset is not animated;
-            if an invalid file format for the endpoint is passed; or if a `size`
-            is passed but the route is not sizable.
+            If an invalid file format for the endpoint is passed;
+            If an animated format is requested for a static asset.
         ValueError
-            If `size` is specified, but is not an integer power of `2` between
-            `16` and `4096` inclusive or is negative.
+            If `size` is specified but is not a power of two or not between 16 and 4096.
         """
-        file_format = file_format.lower()
+        file_format = file_format.upper()
 
         if file_format not in self.valid_formats:
             raise TypeError(
                 f"{file_format} is not a valid format for this asset. Valid formats are: "
                 + ", ".join(self.valid_formats)
             )
 
-        if "hash" in kwargs and not str(kwargs["hash"]).startswith("a_") and file_format == GIF:
-            msg = "This asset is not animated, so cannot be retrieved as a GIF"
+        if "hash" in kwargs and not str(kwargs["hash"]).startswith("a_") and file_format in [AWEBP, GIF, APNG]:
```

Comment:
```suggestion
        if "hash" in kwargs and not str(kwargs["hash"]).startswith("a_") and file_format in {AWEBP, GIF, APNG}:
```

In general, we want to use sets for `in` checks